src/frontend/src/binder/set_expr.rs

```diff
@@ -194,6 +218,85 @@ impl Binder {
         Ok(())
     }
 
+    /// Check the corresponding specification of the set operation.
+    /// Returns the corresponding column index of the left and right side.
+    fn corresponding(
+        &self,
+        left: &BoundSetExpr,
+        right: &BoundSetExpr,
+        corresponding: Corresponding,
+    ) -> Result<(ColIndexMapping, ColIndexMapping)> {
+        let check_duplicate_name = |set_expr: &BoundSetExpr| {
+            let mut name2idx = HashMap::new();
+            for (idx, field) in set_expr.schema().fields.iter().enumerate() {
+                if name2idx.insert(field.name.clone(), idx).is_some() {
+                    return Err(ErrorCode::InvalidInputSyntax(format!(
+                        "duplicated column name `{}` in a select list of a set operation",
```

Comment:
Better to indicate which side of which operation, instead of "a select list of a set operation"

Reply:
Besides, better to add a test case for this error. I didn't see one.

Reply:
There can be more than 2 side. If we have a `A union corresponding B union corresponding C`. It can be confusing to if B is a left or right side

Reply:
I'm not sure. Would that still be better than nothing, since we can exclude `C`? 🤔

Reply:
It's useful for debugging. But I'm just worried users will be confused about which one is left when they see three unioned query. 

I do think printing the whole column list in error msg makes sense know.